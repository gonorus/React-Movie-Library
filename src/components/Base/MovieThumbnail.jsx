import React from 'react';
import PropTypes from 'prop-types';
import './MovieThumbnail.scss';

import Picture from './Picture';
import { DateToMDY } from '../../helpers/Date';
import { DisplayModal } from './Modal';
import MovieDetail from './MovieDetail';

/**
 * Movie Thumbnail with title
 */
const MovieThumbnail = ({ movie }) => {
  const {
    Title, ReleaseDate, ImagePath, MovieDetails
  } = movie;

  /**
   * should displaying Movie details
   */
  const OnMovieSelected = () => {
    DisplayModal(<MovieDetail url={MovieDetails} />);
  };

  /**
   * handle Enter Key pressed
   * @param {Event} evt
   */
  const OnKeyPressed = (evt) => {
    if (evt.charCode === 13) {
      OnMovieSelected();
    }
  };

  return (
    <div
      className='movie-thumbnail-card'
      role='button'
      tabIndex={0}
      onClick={OnMovieSelected}
      onKeyPress={OnKeyPressed}
    >
      <Picture src={ImagePath} alt={Title} />
      <div className='movie-thumbnail-info'>
        <h3>{Title}</h3>
        <p>{DateToMDY(ReleaseDate)}</p>
      </div>
    </div>
  );
};
MovieThumbnail.propTypes = {
  movie: PropTypes.shape({
    Title: PropTypes.string.isRequired,
    ReleaseDate: PropTypes.objectOf(Date).isRequired,
    ImagePath: PropTypes.string.isRequired,
    MovieDetails: PropTypes.string.isRequired,
  }).isRequired,
};
export default MovieThumbnail;
