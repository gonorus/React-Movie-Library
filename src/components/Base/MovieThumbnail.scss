.movie-thumbnail-card {
  padding: 4px;
  min-width: 15%;
  height: auto;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  gap: 8px;

  .movie-thumbnail-info {
    padding: 0 16px;

    h3 {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    p {
      color: rgba(0,0,0,0.6);
    }

    h3,
    p {
      margin: 0;
      white-space: pre;
    }
  }
}

@media screen and (max-width: 600px) {
  .movie-thumbnail-card {
    min-width: 25%;

    .movie-thumbnail-info {
      padding: 0;

      h3,
      p {
        font-size: 12px;
        text-align: center;
      }
    }
  }
}