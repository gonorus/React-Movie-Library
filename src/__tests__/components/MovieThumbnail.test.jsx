import React from 'react';
import renderer, { act } from 'react-test-renderer';
import MovieThumbnail from '../../components/Base/MovieThumbnail';

test('Movie Thumbnail loaded', () => {
  const movie = {
    Title: 'Movie Title', ReleaseDate: new Date('2021-12-04'), ImagePath: 'http://localhost/example.jpg', MovieDetails: 'http://localhost/details'
  };
  const component = renderer.create(
    <MovieThumbnail movie={movie} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => tree.props.onClick());
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => tree.props.onKeyPress({ charCode: 13 }));
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => tree.props.onKeyPress({ charCode: 0 }));
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
