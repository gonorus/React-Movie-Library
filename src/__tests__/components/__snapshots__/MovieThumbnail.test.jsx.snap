// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Thumbnail loaded 1`] = `
<div
  className="movie-thumbnail-card"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="picture-container"
  >
    <img
      alt="Movie Title"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="http://localhost/example.jpg"
    />
    <div
      className="showbox"
    >
      <div
        className="loader"
        style={
          Object {
            "width": 50,
          }
        }
      >
        <svg
          className="circular"
          viewBox="25 25 50 50"
        >
          <circle
            className="path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="movie-thumbnail-info"
  >
    <h3>
      Movie Title
    </h3>
    <p>
      Dec 4, 2021
    </p>
  </div>
</div>
`;

exports[`Movie Thumbnail loaded 2`] = `
<div
  className="movie-thumbnail-card"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="picture-container"
  >
    <img
      alt="Movie Title"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="http://localhost/example.jpg"
    />
    <div
      className="showbox"
    >
      <div
        className="loader"
        style={
          Object {
            "width": 50,
          }
        }
      >
        <svg
          className="circular"
          viewBox="25 25 50 50"
        >
          <circle
            className="path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="movie-thumbnail-info"
  >
    <h3>
      Movie Title
    </h3>
    <p>
      Dec 4, 2021
    </p>
  </div>
</div>
`;

exports[`Movie Thumbnail loaded 3`] = `
<div
  className="movie-thumbnail-card"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="picture-container"
  >
    <img
      alt="Movie Title"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="http://localhost/example.jpg"
    />
    <div
      className="showbox"
    >
      <div
        className="loader"
        style={
          Object {
            "width": 50,
          }
        }
      >
        <svg
          className="circular"
          viewBox="25 25 50 50"
        >
          <circle
            className="path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="movie-thumbnail-info"
  >
    <h3>
      Movie Title
    </h3>
    <p>
      Dec 4, 2021
    </p>
  </div>
</div>
`;

exports[`Movie Thumbnail loaded 4`] = `
<div
  className="movie-thumbnail-card"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="picture-container"
  >
    <img
      alt="Movie Title"
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="http://localhost/example.jpg"
    />
    <div
      className="showbox"
    >
      <div
        className="loader"
        style={
          Object {
            "width": 50,
          }
        }
      >
        <svg
          className="circular"
          viewBox="25 25 50 50"
        >
          <circle
            className="path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="movie-thumbnail-info"
  >
    <h3>
      Movie Title
    </h3>
    <p>
      Dec 4, 2021
    </p>
  </div>
</div>
`;
